feat: Create; add candidate, candidate table, candidate detail, edit …

hatdogguldo opened the pull request:
fixed #137 :    Change the regex validInput

fixed #138 :  Remove error message to the select fields in add candidate form
        
fixed #139:  Remove error message to the select fields in add candidate form

fixed #140 : [I created a php script for adding new column named "candidate_creation"](https://drive.google.com/drive/folders/1tKJrlJ9c2YdfNwNNtgLMUgcDOk-FRHtz?fbclid=IwAR2OB3lpLjHFirKu7OI_LAEu1Jt53dfxtAdG2032GfDB0lROGn8hSms1uiw)

close #141

hatdogguldo committed:
feat: Create; add candidate, candidate table, candidate detail, edit candidate.

biellamariscotes committed:
Merge branch 'development' into CAM

biellamariscotes commented on src/images/candidate-profile/BPF.drawio.png:
Kindly remove the test data.

biellamariscotes commented on src/images/candidate-profile/BPF_Final.drawio.png:
Kindly remove the test data.

biellamariscotes commented on src/images/candidate-profile/Proposed Business Flow Diagram_page-0001.png:
Kindly remove the test data.

biellamariscotes commented on src/images/candidate-profile/Proposed ERD_page-0001.jpg:
Kindly remove the test data.

biellamariscotes commented on src/images/candidate-profile/output-0000.jpg:
Kindly remove the test data.

biellamariscotes commented on src/images/candidate-profile/sad_group3_2024-05-25_07.39pm.png:
Kindly remove the test data.

biellamariscotes commented on src/manage-candidate.php:
```diff
@@ -0,0 +1,340 @@
+
+					<div class="container">
+						<div class="row justify-content-center">
+							<!-- VERIFIED TABLE -->
```
Maintain proper documentation of comments. Replace as "CANDIDATES TABLE".

biellamariscotes commented on src/submission_handlers/update-candidate.php:
```diff
@@ -0,0 +1,54 @@
+        // Handle file upload
+        $photo_url = '';
+        if (isset($_FILES['photo']['name'])) {
+            $target_dir = __DIR__ . '/../images/candidate-profile/';
```
The file path for the images of the candidates is incorrect. These images should be stored in the following directory: `src/user_data/orgname/candidate_imgs`.

hatdogguldo replied:
This part is already fixed.

Now this is my target_dir:
                                       '/../user_data/' . $org_name . '/candidate_imgs/'

biellamariscotes commented:
### ❗ Manage Candidates
- The page seems to be not showing me anything. Take a look:


https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/3739ed5d-b600-403e-9d14-15ac7717fa2a

<hr>


### Add Candidates - Issues Identified

- Name Entry: It does not allow me to enter first names with second names (e.g., Marie Jeremie). An exception should be made for this.
- Position Selection: An error message appears even after selecting a position.
- Block Section: An error message appears even after selecting a block section.


https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/90e18083-bd26-4596-b663-f079569cd839

![Screenshot 2024-06-02 170249](https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/f860078c-431d-4584-9d26-2c4bfaf79e4b)

These are critical bugs as they prevent the submission, so I cannot completely simulate the submission of the form.

I've created an issue ticket for this. Please refer to the linked issue in this PR.

<hr>

Prioritize addressing the bugs, particularly the broken page in Manage Candidates. Once that's resolved, I'll conduct a more thorough review and provide additional feedback on any critical issues if both of Candidate pages are functional. Thank you for your attention to this matter.

### Notes:
- Please avoid pushing test data (like the candidate images in your PR) in the commits.
- The excessive use of `!important` in your styles is concerning as it may override other styles unintentionally. Please use `!important` more sparingly to prevent this.

biellamariscotes committed:
Merge branch 'development' into CAM

hatdogguldo commented:
> ### ❗ Manage Candidates
> * The page seems to be not showing me anything. Take a look:

I created alter-table for candidate to add new key "candidate_creation".  
[Link](https://drive.google.com/drive/folders/1Ke1F_vxh0vj1zzWkpMjmeKA96uUizTsT?fbclid=IwAR2OB3lpLjHFirKu7OI_LAEu1Jt53dfxtAdG2032GfDB0lROGn8hSms1uiw)

hatdogguldo committed:
fix: AT5-101 corrected the validations input on add candidate input fields.

hatdogguldo committed:
Merge branch 'CAM' of https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem into CAM

hatdogguldo committed:
Merge branch 'development' of https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem into CAM

biellamariscotes committed:
Merge branch 'development' into CAM

biellamariscotes committed:
Merge branch 'development' into CAM

biellamariscotes commented:
<h2> Issues Identified</h2>

### Add Candidate
- Election Year is not being inserted into the database.

https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/73e88700-356a-4846-8e24-c00e4dd9b57a

### Manage Candidate
- Although I can see the codes in your files, the functionalities (sort, filter, search, pagination) are not working properly.

https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/assets/135019439/4b062bd9-30de-431f-a1e2-0482fefc7f3f

<hr>

### ❗ Need to be addressed
- Remove the images on src/images/candidate-profile.
- Add the fixed issues (137, 138, 139, 140, 141) on this PR's description so that the included issues on this PR can be automatically done. Refer to this [PR's description ](https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/pull/117)on how to put the fixes.

<hr>

### Additional Notes
- In your **submission_handlers** files, you can use the existing classes in `query-handler.php` (located in `includes/classes`) to fetch data from the database.
- Make sure to also add the codes for the loader in your pages. Refer [here](https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/pull/156).

<hr>

After the **Needs to be addressed** have been done, we can merge this. You can address the identified issues, as well as other enhancements in additional notes, on your future PRs. I will just ticket issues to track these bugs.

Most of the problems are inconsistency with the UI, validations in the form, functionalities of the table— so kindly improve them.

biellamariscotes committed:
Merge branch 'development' into CAM

biellamariscotes commented:
@hatdogguldo Just following up on the requested changes. Please address them as soon as possible. Thanks!

biellamariscotes committed:
Merge branch 'development' into CAM

hatdogguldo committed:
Merge branch 'development' of https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem into CAM

hatdogguldo committed:
fix: input validations, add new sql parameter, and candidate table header.

Carl-Tabuso committed:
Merge branch 'development' into CAM

hatdogguldo committed:
Merge branch 'CAM' of https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem into CAM

hatdogguldo commented:
fixed #137 :    Change the regex validInput

fixed #138 :  Remove error message to the select fields in add candidate form
        
fixed #139:  Remove error message to the select fields in add candidate form

fixed #140 : [I created a php script for adding new column named "candidate_creation"](https://drive.google.com/drive/folders/1tKJrlJ9c2YdfNwNNtgLMUgcDOk-FRHtz?fbclid=IwAR2OB3lpLjHFirKu7OI_LAEu1Jt53dfxtAdG2032GfDB0lROGn8hSms1uiw)

close #141

hatdogguldo committed:
fix: remove inserted files
